documentation/authenticatedoc.md

```diff
@@ -1,206 +1,49 @@
-}
-```
-Description: password cannot be empty string
-### Authentication error code:
-- BAD_CREDENTIALS
-
-##### BAD_CREDENTIALS error description:
-Structure:
-```
-{
-    "status": 401,
-    "code": "BAD_CREDENTIALS",
-    "title": "bad credentials"
-}
-```
-Description: server cannot authenticate user due to invalid name or password
-
-### HTTP status for error responses
-If one of the critical errors occurs http status will be 400 \
-If bad credentials error occurs http status will be 401 \
-Else http status will be 422
-
-### Example of error responses
-##### Without crirital errors
-JSON send:
-```
-{
-    "name": "",
-    "password": "ImF1u9n5k^yPassword"
-}
-```
-Response:
-```
-{
-    "errors": [
-        {
-            "status": 422,
-            "code": "NAME_MISSING",
-            "title": "parameter name is not present"
-        }
-    ],
-    "metaData": null
-}
-```
-##### With crirital errors
-JSON send:
-```
-{
-    "name": null,
-    "password": ""
-}
-```
-Response:
-```
-{
-    "errors": [
-        {
-            "status": 400,
-            "code": "NAME_NULL",
-            "title": "parameter name is null"
-        },
-        {
-            "status": 422,
-            "code": "PASSWORD_MISSING",
-            "title": "parameter password is not present"
-        }
-    ],
-    "metaData": null
-}
-```
-##### With bad credentials
-JSON send:
-```
-{
-    "name": "Monday",
-    "password": "123"
-}
-```
-Response:
-```
-{
-    "errors": [
-        {
-            "status": 401,
-            "code": "BAD_CREDENTIALS",
-            "title": "bad credentials"
-        }
-    ],
-    "metaData": null
-}
-```
+# User authenticate documentation
+
+## List of all endpoints:
+
+| Id |URL|Description|Reference|
+|:-:|:-:|:-:|:-:|
+|1|`/api/authenticate`|Endpoint for user authentication|[click](#apiauthenticate-endpoint)|
+
+
+## api/authenticate endpoint
+| Title | User logout  |
+|:-:|:-|
+| __URL__  | `/api/authenticate` |
+| __Method__    | POST      |
+| __URL PARAMS__ | none      |
+|__Data Params__| Required json:<pre>{<br>&#9;"name": "user_name",<br>&#9;"password": "user_password"<br>}<pre>|
+|__Notes__|none |
+
+### Success response
+|Title|Success response|
+|:-:|:-|
+|Code|200|
+|Body| <pre>{<br>&#9;"data": {<br>&#9;&#9;"token":"jwt_token",<br>&#9;&#9;"name": "user_name",<br>&#9;&#9;"email":"user_email"<br>&#9;},<br>"metaData": null<br>}</pre>|
+|Credentials| Cookie with token |
+|Notes| Be sure to allow credentials from client side |
+
+### Error response
+|Title|Error response|
+|:-:|:-|
+|Code|400 or 401 or 422|
+|Body|<pre>{<br>&#9;"errors": [<br>&#9;&#9;{<br>&#9;&#9;&#9;"status": 400,<br>&#9;&#9;&#9;"code": "error code",<br>&#9;&#9;&#9;"title": "short description of problem"<br>&#9;&#9;}<br>&#9;],<br>&#9;"metaData": null<br>}</pre>|
+|Notes| Table can have multiple elements. List of possible error with short description is presented below |
+
+#### Critical  error codes list
```

Comment:
Again: nie wiem co znaczą "krytyczne" błędy, a co "normalne"?
Czy nie lepiej byłoby zwrócić tą samą informację, niezależnie od tego czy `name` było pustym stringiem, nullem, czy w ogóle nie było tego pola?

Reply:
Zwracanie NAME_NULL oznacza zazwyczaj błąd w wysyłanym JSON'ie. (patrz 40 linijka pliku authenticatedoc.md)
Rozdzielenie NAME_NULL i NAME_MISSING ma głównie na celu łatwiejsze lokalizowanie ewentualnego problemu. 
NAME_NULL: złe typowanie JSON (dostajesz ten error zamiast 500)
NAME_MISSING: połączenie jest nawiązane. Serwer dostaje pusty string. 
W sumie teraz zauważyłem, że niektóre rzeczy są niepotrzebnie wrzucane do tabel. 
Postaram się naprawić to potem.